### PR TITLE
V7 Add the new predefined class types

### PR DESCRIPTION
**Pull request by RexJaeschke**

### CODE REVIEW

File: spec/14-classes.md [hunk 1]
```diff
@@ -1913,6 +1941,34 @@ case, the `Closure` object is empty.
 
 Closure objects can not be serialized or unserialized.
 
+###Class `DivisionByZeroError`
+
+An instance of this class is thrown when an attempt is made to divide a number by zero using the remainder operators ([`%`](10-expressions.md#multiplicative-operators) and [`%=`](10-expressions.md#compound-assignment)). The class type is defined, as follows:
```
Comment (nikic) [inline spec/14-classes.md]: It can also be thrown by `intdiv`, but I guess that's outside the scope of the spec?

File: spec/14-classes.md [hunk 2]
```diff
@@ -1994,4 +2064,22 @@ base class. An instance of this type is automatically created when a
 non-object is [converted to an object](08-conversions.md#converting-to-object-type), or the [member selection
 operator](10-expressions.md#member-selection-operator) is applied to `NULL`, `FALSE`, or an empty string.
 
+###Class `TypeError`
+
+An instance of this class is thrown when any of the following occurs in [strict mode](13-functions.md#type-check-modes):
```
Comment (nikic) [inline spec/14-classes.md]: The first two bullets also apply to weak mode (coercion semantics notwithstanding).

File: spec/14-classes.md (hunk 1 above)
Comment (RexJaeschke) [inline spec/14-classes.md]: Correct; as we're no documenting the library, no mention of that is needed.

File: spec/14-classes.md [hunk 3]
```diff
@@ -1913,6 +1941,34 @@ case, the `Closure` object is empty.
...
+
+###Class `Error`
+
+This class is the base class for all internal PHP error exceptions. It is defined, as follows:
+
+```PHP
+class Error implements Throwable
+{
+}
+```
```
Comment (nikic) [inline spec/14-classes.md]: This should probably mirror the same members we list for `Exception`. (The classes are pretty much the same, only the name differs).

Comment (smalyshev) [inline spec/14-classes.md]: I'm not sure whether we need to duplicate, implementing Throwable should already provide that. Also, I see it's documented incorrectly in the manual - all its methods are final, as in exception (in fact, they use the same function table).

**Comment by smalyshev** [comment]
I think it might be good to put error classes in a separate section.

File: spec/14-classes.md (hunk 3 above)
Comment (nikic) [inline spec/14-classes.md]: @smalyshev The interface does not include properties. The protected ones are part of the API.

Comment (smalyshev) [inline spec/14-classes.md]: Good point, I'll add properties.

**Pull request merged**